src/misc.ts

```diff
@@ -559,3 +560,24 @@ export const getByUnmatchingId =
 
 export const findById = <T extends Node>(id: string, list?: T[]) =>
   list?.find(getById(id));
+
+export const getStatusColor = (
+  status: "error" | "warning" | "info" | "success" | "generic",
+  currentTheme?: DefaultTheme,
+) => {
+  switch (status) {
+    case "error":
+      return "surfaceCriticalDepressed";
+    case "info":
+      return "surfaceBrandDepressed";
+    case "success":
+      return "decorativeSurfaceSubdued2";
+    case "warning":
+      // TODO: use color from new macaw theme when will be ready
+      return currentTheme === "defaultDark" ? "#3E2F0A" : "#FBE5AC";
+    case "generic":
+      return "surfaceBrandSubdued";
+    default:
+      return "surfaceBrandSubdued";
+  }
```

Comment:
**suggestion**: you can set the explicit return type based on the macaw tokens, which will prevent us using a color that does not exist within the token (especially in the future)